modules/gis/src/lib/binary-to-geojson.js

```diff
@@ -63,24 +63,27 @@ function deduceReturnType(dataArray) {
 function parseFeatureCollection(dataArray) {
   const features = [];
   for (const data of dataArray) {
-    let lastIndex = 0;
-    let lastValue = data.featureIds.value[0];
-
-    // Need to deduce start, end indices of each feature
-    for (let i = 0; i < data.featureIds.value.length; i++) {
-      const currValue = data.featureIds.value[i];
-      if (currValue === lastValue) {
-        // eslint-disable-next-line no-continue
-        continue;
+    if (data.featureIds.value.length) {
```

Comment:
Do you need to check that `value` is defined?

```js
data.featureIds.value && data.featureIds.value.length
```
Though I guess it looks like you shouldn't have to according to the declared typing
https://github.com/visgl/loaders.gl/blob/bb450ef5e6f03cdf43269abd5e866274af379ef1/modules/gis/src/types.d.ts#L36

Personally I prefer a guard clause instead of an extra indentation block, so I'd use
```
if (data.featureIds.value.length === 0) {
  continue
}
```
though given you have to add a `no-continue` ignore, maybe that's against the project's style? I don't know. And then you won't have a diff for the rest of this block.

Reply:
I've took the typing of BinaryFeaturesData in mind for this and checked with the output of geojsonToBinary for an empty features array for this check, so that is whay I didn't check that `value` is defined, but I can add it if you think that it is needed.
For the guard clause, as long as we already have another just a few lines below, I think that your suggestion is more consistent with the rest of the function, so I'll apply it.